### ISSUE TITLE
Segmentation faults when using SwiftyJSON as a subproject

### ISSUE THREAD

**Issue by Noobish1** [SwiftyJSON/SwiftyJSON#67]
If you add SwiftyJSON by dragging in the SwiftyJSON.xcodeproj then linking the framework and add a class like the following:
```
import SwiftyJSON

class Object {
func method(json: JSON) {
}
}
```

You'll get a segmentation fault

**Comment by blackbumer** [comment]
@konstantinkoval thanks,
Your hack did the trick. With unit tests where main app goes as target decency importing SwiftyJSON as a file in main app won't help

**Comment by rustedivan** [comment]
Hi all,
I dug into this and have submitted a minimal repro case in [PR to /practicalswift/swift-compiler-crashes](https://github.com/practicalswift/swift-compiler-crashes/pull/47). It seems the compiler crashes when it hits the default arguments (opt: NSJSONReadingOptions = .AllowFragments) in the JSON initialiser. When a initialization like that is optimised out, the compiler segfaults. It's definitely a bug in swiftc.

It only happens when SwiftyJSON is imported as a module, so you can use the .swift file directly in the meantime.
Another workaround is to make sure the call isn't optimised out (e.g. the .cancel() and println(".") workarounds above.

For SwiftyJSON library code, see if you can omit the default arguments for opt: and errorPointer: somehow. Sucks, but swiftc is still, um, a little wonky and this won't probably be fixed for some time.

**Comment by raphaelmor** [comment]
Hi all,
just a small note to say I use SwiftyJSON.swift directly in a Framework, and it builds fine, but it crashes when I run unit tests (seg fault 11 too). I guess it's because doing that Swifty gets in a module (even if it's not his own) and it is the same bug in swiftc.

**Comment by Blackjacx** [comment]
I have a segfault 11 too when I just include SwiftyJSON.swift in my project. It occurred suddenly after a small refactoring. Before everything was finde. So I think it has nothing to do with the subproject but with some strange swift compiler behaviour.

**Comment by endel** [comment]
Same issue here. I'm having a new surprise at each attempt to compile this. 

When registering Alamofire request callbacks, that's the compiler error I was receiving:
```
Command failed due to signal: Segmentation fault: 11
```

After tried what @yuhua-chen suggested, the problem changed, in a very surprising way:

```
Undefined symbols for architecture x86_64:
  "__TMdV10SwiftyJSON4JSON", referenced from:
      __TFFC7example14ViewController11viewDidLoadFS0_FT_T_U0_FV10SwiftyJSON4JSONT_ in ViewController.o
ld: symbol(s) not found for architecture x86_64
```

After removing the Alamofire callback, the program do compile, which doesn't help.

**Comment by rustedivan** [comment]
@endel: would you be able to share a gist of your crashing/non-crashing code? That sounds like a different segfault that I'd like to minimise and report against swiftc asap. Thanks.

**Comment by endel** [comment]
@rustedivan sure, here it is: https://gist.github.com/endel/6e2f9ac7279a6ee42b90

I've written 3 examples to demonstrate what is happening for me:
1. [Runs fine](https://gist.github.com/endel/6e2f9ac7279a6ee42b90#file-example-1-swift)
2. [Segmentation fault](https://gist.github.com/endel/6e2f9ac7279a6ee42b90#file-example-2-swift)
3. [Linker error](https://gist.github.com/endel/6e2f9ac7279a6ee42b90#file-example-3-swift)

I'm using a custom [Request](https://gist.github.com/endel/6e2f9ac7279a6ee42b90#file-request-swift) class for this.

**Comment by rustedivan** [comment]
Excellent, thanks for putting in the effort with the different cases. Swift compiler crashes are collecting over at practicalswift/swift-compiler-crashes so these reports are golden.

**Comment by Blackjacx** [comment]
For me the following caused a segfault: 

    switch type {
		case .aType:
			// Do Something
		default:
			break
	}

and this solved it:

    switch type {
		case .Some(.aType):
			// Do Something
		default:
			break
	}

**Comment by sman591** [comment]
@kostiakoval's workaround solved it for me. I pushed a copy of the changes to  https://github.com/sman591/SwiftyJSON/tree/compiler-segfault-workaround in case anyone else wants to try it.

**Comment by jzarzuel** [comment]
Same here :-(
If you remove "options" parameter from init() and pass "AllowFragments" as default value to NSJSONSerialization (cannot be modified as removed as input parameter)... It works.

It's a shame that Apple had relased this faulty compiler (this is one of the many bugs it has)

**Comment by sferoze** [comment]
I just had this issue! It was hard to track down but essentially once I removed a function that included JSON object from SwiftyJSON class, it crashed my code with the segmentation fault. 

So basically if I removed code from the project it crashed. 

When I imported SwiftyJSON.swift into my project it started to work.

Here was the stack overflow I posted: http://stackoverflow.com/questions/28623405/why-can-i-not-remove-an-unused-function-from-my-swift-class

**Comment by broglep-koubachi** [comment]
We got feedback from apple, they request to test it with Xcode 6.3 and the news swift 1.2 compiler. We still have to migrate our project, so we can't yet test it, so if someone else that has the problem could try it with Xcode 6.3 that would be great

**Comment by imanzarrabian** [comment]
@broglep-koubachi  I had the exact same pb combining Alamofire and SwiftyJSON doing something like this        
Alamofire.request(.GET, APIClientConstants.kApiBaseUrl + "/vls/v1/stations", parameters: params).responseJSON { (req, response, json, error) in
            if (error != nil) {
                completionHandler(APIResponseState.Error(error!))
            }
            else {
                var json = JSON(json!)
                completionHandler(APIResponseState.Correct(json))
            }
        }
It wouldn't compile under Xcode 6.1.1 and swift 1.1 but it works with swift 1.2 and Xcode 6.3 after manually updating Alamofire and SwiftyJSON to Swift 1.2 of course

**Comment by danielkiedrowski** [comment]
+1 Removing the options parameter from init(data:options:error:) worked for me as well.

**Comment by Isuru-Nanayakkara** [comment]
This seems to be fixed in Xcode 6.3.
